clients/vault/src/oracle/types.rs

```diff
@@ -134,50 +140,58 @@ impl<K: PartialEq + Debug, T> Default for LimitedFifoMap<K, T> {
 	}
 }
 
-/// The slot is not found in the `StellarMessage::TxSet(...)`, therefore this map
-/// serves as a holder of the slot when we hash the txset.
 #[derive(Clone)]
-pub struct TxSetHashAndSlotMap {
-	hash_slot: HashMap<TxSetHash, Slot>,
-	slot_hash: HashMap<Slot, TxSetHash>,
+pub struct DoubleSidedHashMap<T1, T2> {
+	t1_to_t2_map: HashMap<T1, T2>,
+	t2_to_t1_map: HashMap<T2, T1>,
 }
```

Comment:
I really like the idea, just the naming of `T1` and `T2` is a bit lacking. How about we just use some letters and no number?
Something like `<K, V>` and then we have `k_to_v_map` and `v_to_k_map`. K and V would loosely resemble key and value (which does not make perfect sense but at least has a familiar ring to it).

Reply:
updated